frontend/src/app/useApplicationSettings.tsx

```diff
@@ -59,3 +61,27 @@ export const useApplicationSettings = (): {
 
   return { dashboardConfig: retConfig, loaded, loadError };
 };
+
+export const useClusterStorageClasses = (): {
+  clusterStorageClasses: StorageClassKind[];
+} => {
+  const defaultStorageClass: StorageClassKind[] = [];
+  const [clusterStorageClasses, setClusterStorageClasses] = React.useState(defaultStorageClass);
+  React.useEffect(() => {
+    const myClusterStorageClasses = async () => {
+      const clusterStorageClasses = await getStorageClasses()
+        .then((res: StorageClassKind[]) => res)
+        .catch(() => {
+          const scEmptyArray: StorageClassKind[] = [];
+          return scEmptyArray;
+        });
+      setClusterStorageClasses(clusterStorageClasses);
+      return myClusterStorageClasses;
+    };
+
+    // fetch data inside useEffect
+    myClusterStorageClasses();
+  }, []);
+
+  return { clusterStorageClasses };
+};
```

Comment:
I think you want to move this over to our infra for fetching data.

Create a file in `frontend/src/concepts/k8s` called `useStorageClasses.ts`, and make it look something like this:
```
const useStorageClasses = () =>
  useFetchState(getStorageClasses, []);
  
export default useStorageClasses;
```

This pattern allows you to not worry about state; loaded, error, or data. It's all built into the singular `useFetchState` utility we have.

You can use it like:
```
const [storageClasses, isLoaded, error] = useStorageClasses(); 
```
Your flow may just want the first element, `storageClasses`. Not sure we need to worry too much about error states or loaded states as it will be used later adhoc.

Reply:
yeah, I was wondering about the useFetchState utility, think I understand better now, good point.

Reply:
in that new file, do I need to work with React Callbacks, like  

const callback = React.useCallback<FetchStateCallbackPromise<SecretKind | null>>(

similar to https://github.com/opendatahub-io/odh-dashboard/blob/5a2b13d7a95706c448a425b847aaa6512f536ef7/frontend/src/concepts/pipelines/elyra/useElyraSecret.ts#L6  ?

Reply:
So you need to make a stable reference... `const myFunc = () => {}` in the root of a React function will just not work. Because we want to retrigger the calls off of the reference changes.

So you either need to use a module import (these are computed once at first import) or you need to use a `useCallback` and set up the dependencies `[...]` to be items that if they change, you want to refetch. Like input variables `foo` changes to `bar` I may want to retrigger a new fetch for `bar`. 

For your use-case I suggested `useFetchState(getStorageClasses, []);` because `getStorageClasses` needs no input, and is a module exported function, so it's stable referenced.

Hopefully that makes sense.